frozen-abi/macro/src/lib.rs

```diff
@@ -395,19 +409,26 @@ fn frozen_abi_enum_type(input: ItemEnum, expected_digest: &str) -> TokenStream {
 #[cfg(RUSTC_WITH_SPECIALIZATION)]
 #[proc_macro_attribute]
 pub fn frozen_abi(attrs: TokenStream, item: TokenStream) -> TokenStream {
-    let args = parse_macro_input!(attrs as AttributeArgs);
     let mut expected_digest: Option<String> = None;
-    for arg in args {
-        match arg {
-            NestedMeta::Meta(Meta::NameValue(nv)) if nv.path.is_ident("digest") => {
-                if let Lit::Str(lit) = nv.lit {
-                    expected_digest = Some(lit.value());
-                }
-            }
-            _ => {}
+    let attrs_parser = syn::meta::parser(|meta| {
+        if meta.path.is_ident("digest") {
+            expected_digest = Some(meta.value()?.parse::<LitStr>()?.value());
         }
-    }
-    let expected_digest = expected_digest.expect("the required \"digest\" = ... is missing.");
+        Ok(())
+    });
+    parse_macro_input!(attrs with attrs_parser);
+
+    let expected_digest = match expected_digest {
```

Comment:
nit: seems the fancy `let else` can be used?

Reply:
Applied.